sstables/sstables.cc

```diff
@@ -3471,6 +3475,7 @@ future<> sstable::destroy() {
     if (_cached_index_file) {
         co_await _cached_index_file->evict_gently();
     }
+    co_await _storage->destroy(*this);
```

Comment:
An exception thrown from here can hide a previous error in `close_files()`. Speaking of which, that does sound like something specific to the local storage...

Reply:
Yes, that's true. However
1. filesystem_storage::destroy cannot return exceptional future
2. s3_storage::destroy calls s3_client.close() which in turn calls http::client::close() which in turn closes in and out for a socket and .close() for streams is :crossed_fingers: not supposed :four_leaf_clover: to return exceptions
3. clients live as "values" on sstables until #13458 , in which the plan is to move all clients somewhere else and stop making sstables close them upon destroy

Reply:
@denesb , do you agree with my justification of risky exception handling in this place? (not a merge ping, it has conflicts and the rebase is needed anyway)

Reply:
I'm fine with a "looks dangerous" but "works with the current code" as long as there is a plan to address this later, and as I understood this close will go away from sstable desctructors soon. So I'm fine with it.